### DISCUSSION TITLE
Issues using tiktok-whisper on windows

### DISCUSSION THREAD

**Discussion by Elemode451** [daymade/tiktok-whisper#4] [Q&A] [answered]
hello,
I'm working on windows and cannot seem to get the software to work.

my error looks like this:

2024/02/09 20:26:55 Running transcription command...
 command: /Users/tiansheng/workspace/cpp/whisper.cpp/main -m /Users/tiansheng/workspace/cpp/whisper.cpp/models/ggml-large-v2.bin --print-colors -l zh --prompt 以 下是简体中文普通话: -otxt -f C:\Users\thest\OneDrive\Desktop\NDSAssembly\tiktok-whisper-main\tiktok-whisper-main\data\mp3\ndsTest\garytherccar_1700342215_7302914097308273953_16khz.wav -of ./1
2024/02/09 20:26:55 Error running transcription command: exec: "/Users/tiansheng/workspace/cpp/whisper.cpp/main": executable file not found in %PATH%
2024/02/09 20:26:55 Failed to insert data into database: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub

With that being said, I tried fixing the whisper error and the CGO error, but it seems like the program is trying to use a different user's whisper. i cannot figure out how to fix this. Any ideas?

**Comment by daymade** [comment, marked as answer]
hi elemode:

please follow the updated instruction in readme:

1. change your path to whisper_cpp.
    tiktok-whipser is based on two whisper engine: local whisper_cpp and remote openai whisper api. you should modify the binaryPath and modelPath in `func provideLocalTranscriber()` of  `tiktok-whisper/internal/app/wire.go` manually 
    ```go
    func provideLocalTranscriber() api.Transcriber {
        // Modify binaryPath and modelPath to your paths here!
        binaryPath := "~/workspace/cpp/whisper.cpp/main"
        modelPath := "~/workspace/cpp/whisper.cpp/models/ggml-large-v2.bin"
        return whisper_cpp.NewLocalTranscriber(binaryPath, modelPath)
    }
    ```
2. then regenerate wire configration
    ```shell
    cd ./internal/app
    go install github.com/google/wire/cmd/wire@latest
    wire
    ```

**Reply by Elemode451** [reply]
hey daymade,
Thank you so much for the help!
I just wanted to check in on one thing - maybe I am missing something easy, but I don't actually have a 'main' in whisper. I have installed everything required of whisper, but it still doesn't work. I'm geting only one error now - it was there before as well. 

I get an error of "Error running transcription command: exec: "/Volumes/SSD2T/workspace/cpp/whisper.cpp/main": executable file not found in %PATH%"

Is this a tiktokwhisper issue or an issue in actually building whisper itself?
Thank you!

**Reply by daymade** [reply]
actually `main` is the executable program from the [whisper.cpp](https://github.com/ggerganov/whisper.cpp) project, this error message does not indicate an issue of tiktok-whisper, you can fix it by building whipser.cpp on your machine.

please follow the quickstart to build out a `main`: https://github.com/ggerganov/whisper.cpp?tab=readme-ov-file#quick-start
